src/platformAccessory.ts

```diff
@@ -66,6 +68,17 @@ export class SamsungACPlatformAccessory {
       .setProps(temperatureProps)
       .onSet(this.handleCoolingTemperatureSet.bind(this))
       .onGet(this.handleHeatingTemperatureGet.bind(this));
+
+    /**
+     *  Humidity Service if enabled
+     */
+    if (this.platform.config.showHumidity) {
+      this.humidityService = this.accessory.getService(this.platform.Service.HumiditySensor)
+        || this.accessory.addService(this.platform.Service.HumiditySensor);
+
+      this.humidityService.getCharacteristic(this.platform.Characteristic.CurrentRelativeHumidity)
+        .onGet(this.handleCurrentHumidityGet.bind(this));
```

Comment:
What do you think about create a private function like `enableHumidityService` and just call the function on the constructor?